File: programmers-coding-test/Python/Level1/체육복.py
def solution(n, lost, reserve):
    clothes = [1] * (n + 1)
    for i in lost:
        clothes[i] -= 1
    for i in reserve:
        clothes[i] += 1
    for i in range(1, n + 1):
        if clothes[i] == 2:
            if clothes[i - 1] == 0:
                clothes[i] -= 1
                clothes[i - 1] += 1
            elif i < n:
                if clothes[i + 1] == 0:
                    clothes[i] -= 1
                    clothes[i + 1] += 1
    return n - clothes.count(0)

File: programmers/C++/Level1/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {

    int suit_count = 0;
    vector<int> suit(n, 1);  // 1로 채워진 n개의 vector를 생성

    for (int i : lost) suit[i - 1]--;
    for (int i : reserve) suit[i - 1]++;

    for (int i = 0; i < suit.size(); i++)
    {
        if (suit[i] == 0)
        {
            if (i > 0 && suit[i - 1] > 1)  // 앞 친구에게 체육복을 빌리는 경우
            {
                suit[i - 1]--;
                suit[i]++;
            }
            else if (i < suit.size() - 1 && suit[i + 1] > 1)  // 뒷 친구에게 체육복을 빌리는 경우
            {
                suit[i + 1]--;
                suit[i]++;
            }
        }
    }

    for (int i = 0; i < suit.size(); i++)
    {
        if (suit[i] > 0) suit_count++;
    }

    return suit_count;
}